rest-api-spec/src/main/resources/rest-api-spec/test/indices.recovery/10_basic.yml

```diff
@@ -40,6 +40,56 @@
   - gte:   { test_1.shards.0.verify_index.check_index_time_in_millis:     0                       }
   - gte:   { test_1.shards.0.verify_index.total_time_in_millis:           0                       }
 ---
+"Indices recovery test for closed index":
+  - skip:
+      version: " - 7.99.99"
+      reason: closed indices are replicated starting version 8.0.0
+
+  - do:
+      indices.create:
+        index:  test_2
+        body:
+          settings:
+            index:
+              number_of_replicas: 0
+
+  - do:
+      indices.close:
+        index: test_2
+  - is_true: acknowledged
+
+  - do:
+      cluster.health:
+        index: test_2
+        wait_for_status: green
+
+  - do:
+      indices.recovery:
+        index: [test_2]
+        human: true
+
+  - match: { test_2.shards.0.type:                                 "EXISTING_STORE"        }
```

Comment:
As discussed on another channel, a user could find it odd that closing an index makes it go through recovery. Also, if cluster health will be changed to by default also look at closed indices, cluster health could go red for a short period when closing an index. This is more of a discussion subject: is this external behaviour acceptable or not. Maybe @ywelsch has a view on this?

Reply:
> As discussed on another channel, a user could find it odd that closing an index makes it go through recovery.

I can understand this point of view, but I also think that we could mitigate this with a better documentation. 



> Also, if cluster health will be changed to by default also look at closed indices, cluster health could go red for a short period when closing an index. This is more of a discussion subject: is this external behaviour acceptable or not.

I forgot to mention during our discussion that we talked about this in our team meeting few months ago and we agreed on the fact that this situation is acceptable. This behavior is similar to what happens today when an index is reopened, turning the cluster state to RED until primary shards are assigned.

Reply:
Note that if the red cluster health is too big of a concern, we can change that as well by making it only yellow during this transition (can be done by adapting `ClusterShardHealth#getInactivePrimaryHealth`).